frontend/src/employee-frontend/components/FinancePage.tsx

```diff
@@ -18,90 +20,163 @@ import InvoicesPage from './invoices/InvoicesPage'
+              link: '/finance/fee-decisions',
+              label: i18n.header.feeDecisions
+            }
+          : null,
+        user?.permittedGlobalActions.includes('SEARCH_VOUCHER_VALUE_DECISIONS')
+          ? {
+              id: 'value-decisions',
+              link: '/finance/value-decisions',
+              label: i18n.header.valueDecisions
+            }
+          : null,
+        user?.permittedGlobalActions.includes('SEARCH_INVOICES')
+          ? {
+              id: 'invoices',
+              link: '/finance/invoices',
+              label: i18n.header.invoices
+            }
+          : null,
+        user?.permittedGlobalActions.includes(
+          'SEARCH_VOUCHER_VALUE_DECISIONS'
+        ) && featureFlags.experimental?.voucherUnitPayments
           ? {
               id: 'payments',
               link: '/finance/payments',
               label: i18n.header.payments
             }
           : null,
-        {
-          id: 'income-statements',
-          link: '/finance/income-statements',
-          label: i18n.header.incomeStatements
-        }
+        user?.permittedGlobalActions.includes(
+          'FETCH_INCOME_STATEMENTS_AWAITING_HANDLER'
+        )
+          ? {
+              id: 'income-statements',
+              link: '/finance/income-statements',
+              label: i18n.header.incomeStatements
+            }
+          : null
       ].flatMap((tab) => (tab !== null ? tab : [])),
-    [i18n]
+    [i18n, user]
   )
 
-  return (
-    <>
-      <Tabs tabs={tabs} />
-      <Gap size="s" />
-      <Routes>
-        <Route
-          path="fee-decisions"
-          element={
-            <EmployeeRoute title={i18n.titles.feeDecisions}>
-              <FeeDecisionsPage />
-            </EmployeeRoute>
-          }
-        />
+  const requireOneOfPermittedActionsForPage = (
+    { path, title, Page }: PageProps,
+    ...actions: Action.Global[]
+  ): React.ReactNode => {
+    if (
+      actions.some((action) => user?.permittedGlobalActions.includes(action))
+    ) {
+      return (
         <Route
-          path="value-decisions"
+          key={path}
+          path={path}
           element={
-            <EmployeeRoute title={i18n.titles.valueDecisions}>
-              <VoucherValueDecisionsPage />
+            <EmployeeRoute title={title}>
+              <Page />
             </EmployeeRoute>
           }
         />
-        <Route
-          path="invoices"
-          element={
-            <EmployeeRoute title={i18n.titles.invoices}>
-              <InvoicesPage />
-            </EmployeeRoute>
-          }
-        />
-        {featureFlags.experimental?.voucherUnitPayments && (
-          <Route
-            path="payments"
-            element={
-              <EmployeeRoute title={i18n.titles.payments}>
-                <PaymentsPage />
-              </EmployeeRoute>
-            }
-          />
+      )
+    } else {
+      return null
+    }
+  }
+
+  const permittedRoutes = new Map([
```

Comment:
This thing feels overly complicated. 

- I'm not sure if conditionally including routes is even necessary, since even if the user manages to get there with direct URL, it wont show them anything they don't have access to.
- It looks like the `tabs` and `permitted` routes need to be in sync (e.g. doesn't make sense to render a tab if its associated route isn't permitted) so I think these two lists should be merged so you also don't need to duplicate the role checking

Reply:
I agree with the merging. I was not sure if it would be considered a problem or not, if the user may access the route with direct URL, so i decided to filter them also. I shall make the fix so that only the tabs are filtered according to permissions.